src/lib/Libifl/list_link.c

```diff
@@ -191,6 +189,18 @@ delete_link(struct pbs_list_link *oldp)
 	oldp->ll_prior = oldp;
 }
 
+/**
+ * @brief delete an entry from the list and clear the struct
+ *
+ * @param[in] oldp       ptr to link to delete
+ */
+void
+delete_clear_link(struct pbs_list_link *oldp)
```

Comment:
why do we need this function? Usually the callers call delete_link(x) and then free(x). Why do we need to set the ll_struct to NULL? Is it so this object is ready to be appended/inserted to another list?

Reply:
free_ru will delete all the items in the list. delete_clear_link() will help me in deleting only the current item as it won't proceed after the link. I can also update free_ru to accept a new parameter just to delete the first item, but I feel this should be a capability of the list routines.